### ISSUE TITLE
Imported next-server modules are duplicated, causing duplicate react contexts

### ISSUE THREAD

**Issue by bubba** [vercel/next.js#25303]
### What version of Next.js are you using?

10.2.1

### What version of Node.js are you using?

16.1.0

### What browser are you using?

Chrome and Safari

### What operating system are you using?

macOS

### How are you deploying your application?

None

### Describe the Bug

We use `@apollo/client` to execute some graphQL queries during SSR: we do this in getServerSideProps, and most importantly by rendering the component tree before nextjs actually renders it on the server. A simplified example of how this looks is below:

```typescript
export function getServerSideProps(ctx) {
  // ....
     await renderToStringWithData(
        <RouterContext.Provider value={router}>
          <ApolloProvider client={apolloClient}>
             <Page {...props} />
          </ApolloProvider>
        </RouterContext.Provider >
      )
  // ...
}
```

Most importantly, because we use `useRouter` and render different components based on `query`, we create a dummy next router and pass it into the router provider:

```typescript
  const router = {
    query: ctx.query
  }
```

The router context itself is imported like so:
```typescript
import { RouterContext } from 'next/dist/next-server/lib/router-context'
```

This worked fine up until 10.2.1, where now calls on the server to `useRouter` return `null`.
After doing some debugging it turns out this is caused by the two different `router-context` modules being imported: One in our app, and one in `next/router`.
So whenever `useRouter` calls `useContext`, it looks up a different `RouterContext` than the `RouterContext` we imported, and thus returns null.

Previously in 10.2.0 these imported modules were the same, and we believe the PR that caused this new behaviour is https://github.com/vercel/next.js/pull/24603

The difference between the generated webpack imports is that on 10.2.0, they look like this:

```javascript
/* harmony import */ var next_dist_next_server_lib_router_context__WEBPACK_IMPORTED_MODULE_7__ = __webpack_require__(/*! next/dist/next-server/lib/router-context */ "next/dist/next-server/lib/router-context");
```

But on 10.2.1 it now becomes:

```javascript
/* harmony import */
var next_dist_next_server_lib_router_context__WEBPACK_IMPORTED_MODULE_7__ = __webpack_require__(/*! next/dist/next-server/lib/router-context */
"./node_modules/next/dist/next-server/lib/router-context.js");
```
`./node_modules` has been prefixed

`next/dist/client/router.js` imports the module as `require("../next-server/lib/router-context")`, so understandably webpack gives back two different modules, with two different contexts

### Expected Behavior

Importing modules from `next/dist/next-server/` should resolve to the same module as the modules used by next itself

### To Reproduce

https://github.com/bubba/next-webpack-import-bug.git

**Comment by ijjk** [comment]
Hi, this should be fixed in `v10.2.4-canary.1` of Next.js, please update and give it a try!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
